src/python/pants/backend/scala/subsystems/scalac.py

```diff
@@ -26,17 +26,36 @@ class Scalac(Subsystem):
     plugins_global = StrListOption(
         "--plugins-global",
         help=(
-            "A list of addresses of `scalac_plugin` targets which should be used for "
+            "DEPRECATED: A list of addresses of `scalac_plugin` targets which should be used for "
             "compilation of all Scala targets in a build.\n\nIf you set this, you must also "
             "set `[scalac].plugins_global_lockfile`."
         ),
     ).advanced()
+
+    # TODO: see if we can use an actual list mechanism? If not, this seems like an OK option
+    default_plugins = DictOption[str](
+        "--experimental-plugins",
```

Comment:
Maybe call this "plugins-for-resolve" to match the `--scala-version-for-resolve` naming? Also calls out what is being mapped.